### PR TITLE
Fix activate standby server ignores all errors

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
/kind bug
related #22028

### CODE REVIEW

**Review by congqixia — CHANGES_REQUESTED** [review]
`ProcessActiveStandBy` return error is handled only in `querycoordv2` but not in `Datacoord` and `RootCoord`

File: internal/datacoord/server.go [hunk 1]
```diff
@@ -239,7 +239,9 @@ func (s *Server) Register() error {
 	s.session.Register()
 	if s.enableActiveStandBy {
 		s.icSession.ProcessActiveStandBy(nil)
```
Comment (congqixia) [inline internal/datacoord/server.go:241]: Why ignore this error? Despite of the fact that nil `activateFunc` will not result error return for now. We shall not depend on the detail implementation of `ProcessActiveStandBy` here

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -468,21 +470,27 @@ func (c *Core) Init() error {
...
 				}
 			})
+			if err != nil {
+				return err
+			}
 			c.startOnce.Do(func() {
-				if err := c.startInternal(); err != nil {
-					log.Warn("RootCoord start failed", zap.Error(err))
+				if err = c.startInternal(); err != nil {
+					log.Error("RootCoord start failed", zap.Error(err))
```
Comment (jaime0815) [inline internal/rootcoord/root_coord.go:487]: Should return error if start failed

Comment (yah01) [reply internal/rootcoord/root_coord.go:487]: line 493 returns it

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**